agent/config.m4

```diff
@@ -173,8 +173,20 @@ if test "$PHP_NEWRELIC" = "yes"; then
     NEWRELIC_CFLAGS="${NEWRELIC_CFLAGS} -Wshadow"
     NEWRELIC_CFLAGS="${NEWRELIC_CFLAGS} -Wpointer-arith"
     NEWRELIC_CFLAGS="${NEWRELIC_CFLAGS} -Wcast-qual"
-    NEWRELIC_CFLAGS="${NEWRELIC_CFLAGS} -Werror"
-    NEWRELIC_CFLAGS="${NEWRELIC_CFLAGS} -Wno-typedef-redefinition"
+
+    dnl Get the Compiler Vendor name for additional flags
+    dnl that depend either on clang or gcc (since both are "GNUC")
+    m4_include([ax_compiler_vendor.m4])
+    AX_COMPILER_VENDOR
+
+    if test "x$ax_cv_cxx_compiler_vendor" = "xclang"; then
+      dnl The flag below only works under clang, but not gcc
+      NEWRELIC_CFLAGS="${NEWRELIC_CFLAGS} -Wno-typedef-redefinition"
+      NEWRELIC_CFLAGS="${NEWRELIC_CFLAGS} -Werror"
+    else
+      dnl By contrast, the next flag only works under gcc
+      NEWRELIC_CFLAGS="${NEWRELIC_CFLAGS} -Wstringop-overflow"
```

Comment:
Unfortunately, this flag seems to be causing an[ issue on the 7.0 build](https://github.com/newrelic/newrelic-php-agent/runs/4821711049?check_suite_focus=true).

```
cc: error: unrecognized command line option '-Wstringop-overflow'; did you mean '-Wstrict-overflow'?
```

Reply:
Dang! Oh well. You guys can always put a note on the `README.md` instead of dealing with this PR...